### ISSUE TITLE
style enqueue needs version

### ISSUE THREAD

**Issue by joyously** [WordPress/twentynineteen#601]
The main style sheet is enqueued poorly.
`	wp_enqueue_style( 'twentynineteen-style', get_stylesheet_uri() );`

1) there is no version number
 With no version number, the WP version is used, and so CSS gets stuck in the cache and is a poor user experience. It needs a version number so that changes bust the cache.
2) it uses `get_stylesheet_uri()`
 If a child theme is active, the parent will load the child style and put the parent version number on it. The child theme will then have to load the parent style and version. That's crazy. The parent should just load the parent style with its version number. If you want to, you can load the child style for the child, but the style and version should not be mixed.

 Here is how I do it to load the parent and child styles, and the correct version number for both:
```
$theme = wp_get_theme();
$version = $theme->parent() ? $theme->parent()->get( 'Version' ) : $theme->get( 'Version' );
wp_enqueue_style( 'twentynineteen-style', get_template_directory_uri() . '/style.css', array(), $version );
if ( is_child_theme() ) {
	wp_enqueue_style( get_stylesheet() . '-style', get_stylesheet_uri(),
		array( 'twentynineteen-style' ), $theme->get( 'Version' ) );
}
```

**Comment by peterwilsoncc** [comment]
Related PR #549.

**Comment by joyously** [comment]
I know I saw that one, but was thinking it was the print styles (which already have the version)...

**Comment by kjellr** [comment]
@joyously #549 has been merged — it targets the main styles, not the print styles. Does that take care of this issue? 

Thanks!

**Comment by joyously** [comment]
Not quite. It still uses `get_stylesheet()`, which means that a child theme will have to load the parent styles.

**Comment by kjellr** [comment]
Ok, got it. I'll update the title of the ticket to match.

**Comment by grappler** [comment]
FYI it is a theme review requirement to use `get_stylesheet_uri()`. I previously made a PR to `_s` to fix this issue but it was reverted. https://github.com/Automattic/_s/pull/638

This is not a new problem.

**Comment by joyously** [comment]
Can you point me to where that requirement is written? I've never seen it.
And it was only reverted because there was no discussion before the merge.

As a default theme, Twenty Nineteen is more likely to have child themes made of. It is more likely for a user to want the parent styles than not, so having the parent load the child styles causes more problems for the user than just loading its own styles.
